docs/docs/guide/debugging.mdx

```diff
@@ -0,0 +1,303 @@
+(unavailable for JSC) is Chrome's built-in feature that allows for connecting to a
+remote JavaScript runtime. This means that the code you are debugging actually
+runs on the device itself and behaves the same way as it would with DevTools detached.
+
+- [**Flipper**](https://fbflipper.com/) (Hermes debugger) is an app created by
+Facebook that makes it easy to use the _Chrome DevTools_ while providing additional tools for
+UI inspection and debugging.
+
+- [**Safari DevTools**](https://reactnative.dev/docs/debugging#safari-developer-tools)
+is Safari's built-in feature that is only available for iOS devices running JSC. It works
+very similarly to Chrome DevTools - it allows you to connect to a remote runtime
+and keep code execution on device.
+
+- [**React Developer Tools**](https://reactnative.dev/docs/debugging#react-developer-tools)
+are a standalone app that allows debugging UI through the inspector, as well as monitoring
+performance and profiling your app.
+
+### The JS context vs. the UI context
+
+It is important to understand the differences between the JS and UI contexts.
+One thing to note is that adding Reanimated changes nothing when it comes
+to debugging the regular JS context that is available in all React Native apps.
+It is only debugging the UI context, which is specific to the Reanimated
+library that might be difficult to debug in some scenarios, which are all
+explained in this article.
+
+You can read more about Reanimated's architecture [here](../fundamentals/architecture).
+
+### Debugging web apps
+
+For debugging web apps, you may use the tools you prefer or any that are provided
+by the browser of your choice. We did not discover any issues with debugging
+apps using Reanimated on web.
+
+### Summary
+
+<SummaryTable />
+
+:::caution
+
+Remember that console logs will always appear in the primary JS context as the
+`console.log` function on the UI thread is just a reference to the one from
+the JS thread.
+
+:::
+
+### Chrome Debugger
+
+<ChromeDebuggerTable />
+
+**Summary:**
+Works, but uses web implementations of functions and runs everything on the JS thread.
+
+Since the Chrome Debugger runs its own web worker, all the code is run on
+the JS thread, and it uses the JavaScript engine provided by your web
+browser (V8 in Chrome, JSC in Safari and SpiderMonkey in Firefox). This
+means that this piece of code:
+
+```js
+function runWorklet() {
+  'worklet';
+  console.log('worklet:', _WORKLET);
+}
+runOnUI(runWorklet)();
+```
+
+would output:
+
+```
+LOG: worklet: false
+```
+
+Another side effect is that Reanimated uses web implementations of all
+functions. This means that the `scrollTo` function will work
+(using the native web implementation), but the `measure`
+function will not be available, and its usage will trigger this error:
+
+```
+[reanimated.measure] method cannot be used for web or Chrome Debugger
```

Comment:
This error message has been improved in #3453 and will be changed again in #3598.